src/data/vectorOperationsProvider.ts

```diff
@@ -37,9 +37,9 @@ export class VectorOperationsProvider {
     return this.vectorOperations;
   }
 
-  buildVectorOperationsConfig(config) {
+  buildVectorOperationsConfig(config: IndexConfiguration) {
     const indexConfigurationParameters: ConfigurationParameters = {
-      basePath: config.hostUrl,
+      basePath: config.indexHostUrl,
```

Comment:
I had updated the name of this field before merging the previous PR (https://github.com/pinecone-io/pinecone-ts-client/pull/148), and we apparently didn't have any type safety here as `config` seemed to be `any`. The provider was essentially giving `basePath: undefined` to each instance of `Configuration`. 🤦‍♂️

Reply:
Nice to see some type safety added here. But the `index` part of the `indexHostUrl` name seems a bit redundant if the class is called `IndexConfiguration`